### DISCUSSION TITLE
SaaS Database Design Question

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#17155] [Q&A]
Hi,

As a side project, I am building a real estate SaaS platform.
Each user ,after login, can create a property or be invited (by its owner) to join an existing one as guest. 
The owner of the property can add information about its property (e.g. room description , expense he has related to it ...) and the guest can do it as well ( but on a smaller set of elements). 

Currently, I struggle on designing the db schema from a security access point of view. 
My approach is to create inside the Property model, a property_permissions element. 

model PropertyPermission { 
int id 
writeThis Users[]
writeThat Users[]
readThis Users[]
readThat Users[]
....
}

Would this be a model to consider and wouldn't it be too complicated to check access ? 
Indeed a basic example of action user will do is to list all the properties they have access to. 

I would appreciate any direct help or some redirection as well :)
Thank you so much

**Comment by nurul3101** [comment]
Hello @laurent512 👋 

Your approach of defining PropertyPermissions is valid and can definitely be used. To simplify checking access, you can also consider designing a table called `PropertyUserPermission` which could have these fields:
```
property_id: the ID of the property
user_id: the ID of the user
read: a boolean indicating whether the user has read access to the property
write: a boolean indicating whether the user has write access to the property
```
This approach can be useful if you want to store more granular permissions for each user (e.g. read-only access vs. read-write access). It also allows you to easily add or remove permissions for a particular user without having to update the entire PropertyPermission model.

That being said, whether or not to use a separate `PropertyUserPermission` table will depend on the specific requirements and complexity of your application. If you only need to store basic read and write permissions for each user, the PropertyPermission model you described may be sufficient.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
